readthedocs/projects/tasks.py

```diff
@@ -778,6 +778,24 @@ def symlink_domain(project_pk, domain_pk, delete=False):
             sym.symlink_cnames(domain)
 
 
+@app.task(queue='web')
+def remove_orphan_symlinks():
+    """
+    Remove orphan symlinks.
+
+    List CNAME_ROOT for Public and Private symlinks, check that all the listed
+    cname exist in the database and if doesn't exist, they are un-linked.
+    """
+    for symlink in [PublicSymlink, PrivateSymlink]:
+        for domain_path in [symlink.PROJECT_CNAME_ROOT, symlink.CNAME_ROOT]:
+            valid_cnames = set(Domain.objects.all().values_list('domain', flat=True))
+            orphan_cnames = set(os.listdir(domain_path)) - valid_cnames
+            for cname in orphan_cnames:
+                orphan_domain_path = os.path.join(domain_path, cname)
+                log.info('Unlinking orphan CNAME: %s', orphan_domain_path)
+                os.unlink(orphan_domain_path)
+
```

Comment:
I wonder if these should go in another file. `projects/tasks` is already _huge_, so it might make sense to move this into `core/tasks.py` or somewhere else specifically for scheduled tasks?

Reply:
Yes. I agree with this.

I created this issue (#3775) to refactor this and include some other tasks also --we can discuss in the issue.

Reply:
I actually dislike putting things in `core`. Why not split up into:

* `readthedocs.projects.tasks.build`
* `readthedocs.projects.tasks.sync`
* etc

Some of the tasks might make more sense in `builds/` as well. I find `core/` is not a helpful concept, save for only the most generalized of code.